eth/ethutils/receipt.go

```diff
@@ -56,6 +56,8 @@ func MarshalReceipt(
 
 	if !chainConfig.IsLondon(header.Number.Uint64()) {
 		fields["effectiveGasPrice"] = (*hexutil.Big)(txn.GetPrice().ToBig())
+	} else if txn.Type() == types.DepositTxType {
```

Comment:
Another fix that forces `effectiveGasPrice` for deposit tx to be always 0. This should be properly encapsulated into the transaction type like `txn.GetGasPrice(baseFee)` that returns properly calculated gas price of the tx. But for now, this is just a hot fix.